library/src/main/java/com/getstream/sdk/chat/ImageLoader.kt

```diff
@@ -33,6 +41,43 @@ internal object ImageLoader {
             }
     }
 
+    fun ImageView.loadWithGifSupport(
+        uri: String?,
+        @DrawableRes placeholderResId: Int?,
+        onStart: () -> Unit = {},
+        onCancel: () -> Unit = {},
+        onError: (throwable: Throwable) -> Unit = {},
+        onSuccess: () -> Unit = {},
+    ): RequestDisposable =
+        load(uri, getImageLoaderWithGifSupport(context)) {
+            placeholderResId?.let { placeholder(it) }
+            listener(
+                onStart = { onStart() },
+                onCancel = { onCancel() },
+                onError = { _, throwable -> onError(throwable) },
+                onSuccess = { _, _ -> onSuccess() },
+            )
+        }
+
+    private fun getImageLoaderWithGifSupport(
+        context: Context,
+        // TODO: We should probably allowHardware for performance improvements but we do software rendering in PorterShapeImageView
```

Comment:
Does using software rendering in `PorterShapeImageView` conflict with this? The docs for Coil say to disable hardware only if we run into issues otherwise.

Reply:
Unfortunately yes, we are getting ` java.lang.IllegalArgumentException: Software rendering doesn't support hardware bitmaps` when trying to render image inside `PorterShapeImageView`

Reply:
Ah, okay, ouch. I guess we can revisit this and try to work around it if / when we remove Glide and do more Coil work.